File: 하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    int answer = x;
    int sum = 0;
    while (x > 0){
        sum += x % 10;
        x /= 10;
    }
    return (answer % sum == 0);
}

File: 하샤드 수.py
'''
def solution(x):
    
    char = str(x)
    abc = 0
    
    for i in range(len(char)):
        abc = abc + int(char[i])
        
    return x % abc == 0
'''
"""
#2020.07.01
def solution(x):
    return x % sum(map(int ,list(str(x)))) == 0
"""
# 2022.01.30
def solution(x):
    return x % sum(map(int, (list(str(x))))) == 0
